sdk/aqueduct/integrations/connect_config.py

```diff
@@ -147,18 +147,48 @@ class SlackConfig(BaseConnectionConfig):
     enabled: bool
 
 
-class AWSConfig(BaseConnectionConfig):
-    access_key_id: str
-    secret_access_key: str
-
-
 class _SlackConfigWithStringField(BaseConnectionConfig):
     token: str
     channels_serialized: str
     level: str
     enabled: str
 
 
+class DynamicK8sConfig(BaseConnectionConfig):
+    # How long (in seconds) does the cluster need to remain idle before it is deleted.
+    keepalive: Optional[str]
+    # The EC2 instance type of the CPU node group. See https://aws.amazon.com/ec2/instance-types/
+    # for the node types available.
+    cpu_node_type: Optional[str]
+    # The EC2 instance type of the GPU node group. See https://aws.amazon.com/ec2/instance-types/
+    # for the node types available.
+    gpu_node_type: Optional[str]
+    # Minimum number of nodes in the CPU node group. The cluster autoscaler cannot scale below this number.
+    # This is also the initial number of CPU nodes in the cluster.
+    min_cpu_node: Optional[str]
+    # Maximum number of nodes in the CPU node group. The cluster autoscaler cannot scale above this number.
+    max_cpu_node: Optional[str]
+    # Minimum number of nodes in the GPU node group. The cluster autoscaler cannot scale below this number.
+    # This is also the initial number of GPU nodes in the cluster.
+    min_gpu_node: Optional[str]
+    # Maximum number of nodes in the GPU node group. The cluster autoscaler cannot scale above this number.
+    max_gpu_node: Optional[str]
+
+
+class AWSConfig(BaseConnectionConfig):
+    access_key_id: str
+    secret_access_key: str
+    region: str
+    k8s: Optional[DynamicK8sConfig]
+
+
+class _AWSConfigWithSerializedConfig(BaseConnectionConfig):
+    access_key_id: str
+    secret_access_key: str
+    region: str
+    k8s_serialized: Optional[str]  # this is a json-serialized string of DynamicK8sConfig
```

Comment:
specifically, this is intended to be the serialized version of `AWSConfig.k8s` right? Might be good to make that relationalship clear.